src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/SerializationEventsCache.cs

```diff
@@ -15,15 +16,18 @@ internal sealed class SerializationEvents
         private readonly List<MethodInfo>? _onDeserializingMethods;
         private readonly List<MethodInfo>? _onDeserializedMethods;
 
-        internal SerializationEvents(Type? t)
+        internal SerializationEvents([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type? t)
         {
             _onSerializingMethods = GetMethodsWithAttribute(typeof(OnSerializingAttribute), t);
             _onSerializedMethods = GetMethodsWithAttribute(typeof(OnSerializedAttribute), t);
             _onDeserializingMethods = GetMethodsWithAttribute(typeof(OnDeserializingAttribute), t);
             _onDeserializedMethods = GetMethodsWithAttribute(typeof(OnDeserializedAttribute), t);
         }
 
-        private List<MethodInfo>? GetMethodsWithAttribute(Type attribute, Type? t)
+        private List<MethodInfo>? GetMethodsWithAttribute(
+            Type attribute,
+            // currently the only way to preserve base, non-public methods is to use All
```

Comment:
Is that something the linker plans to fix - if so should this be tracked somehow to fix later?

Reply:
I'm starting to wonder if PrivateMethods should be changed to mean private methods across the entire hierarchy - and not the current reflection semantics which is pretty confusing.

Reply:
> I'm starting to wonder if PrivateMethods should be changed to mean private methods across the entire hierarchy

It's going to keep more stuff so the question is whether it's worth it. Matching reflection semantics currently means we keep exactly what "regular" reflection needs. 

Looking at API usage statistics of `Type.BaseType` could be used as a first approximation - apps that don't use it would definitely not benefit from extending the semantics. Apps that use it _might_. Seems like Apisof.net no longer has usage statistics which is where I used to go looking for that stuff.